cpp/demo/hyperelasticity/main.cpp

```diff
@@ -86,8 +86,8 @@ class HyperElasticProblem
     return [&](const Vec x, Vec)
     {
       // Assemble b and update ghosts
-      std::span<T> b(_b.mutable_array());
-      std::ranges::fill(b, 0.0);
+      std::span b(_b.mutable_array());
+      std::ranges::fill(b, 0);
```

Comment:
Tangential, but do we have a set style on this? Personally I prefer using floating point literals even if a cast from an integer literal is possible. We are also using `T(1)` below.

Reply:
I think our house (`0.0`) style pre-dates supporting different types. I prefer
```cpp
std::ranges::fill(b, 0);
```
to
```cpp
std::ranges::fill(b, T(0));
```
because it's less syntax. I think
```cpp
std::ranges::fill(b, 0.0);
```
is possibly misleading since `T` is not necessarily `double`.